config.c

```diff
@@ -173,6 +174,12 @@ config_init(const char *config_path)
           goto out;
         }
         dev_config->timeout = var_uint;
+      } else if (sscanf((char *)buf, "network.local-ip %64s", var_str) == 1) {
```

Comment:
You read 64 bytes but only strncpy 15 later. Should probably read 15 here at most, then we can strncpy 16 and have a guaranteed 0-byte termination.
(I also didn’t write the project initially, C is not a great choice…)